src/server/helpers/collectionRouteUtils.js

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2020 Prabal Singh
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import * as handler from '../helpers/handler';
+import * as search from './search';
+import _ from 'lodash';
+
+
+export async function makeCollectionCreateOrEditHandler(req, res) {
+	const {UserCollection, UserCollectionCollaborator} = req.app.locals.orm;
+	const isNew = !res.locals.collection;
+	let newCollection;
+	let method;
+	if (isNew) {
+		newCollection = await new UserCollection({
+			editorId: req.user.id
+		});
+		method = 'insert';
+	}
+	else {
+		newCollection = await new UserCollection({id: req.params.collectionId}).fetch({
+			require: true
+		});
+		method = 'update';
+	}
+	newCollection.set('description', req.body.description);
+	newCollection.set('name', req.body.name);
+	newCollection.set('public', req.body.privacy === 'Public');
+	newCollection.set('type', req.body.type);
+	await newCollection.save(null, {method});
+
+	const oldCollaborators = res.locals.collection ? res.locals.collection.collaborators : [];
+	const newCollaborators = req.body.collaborators ? req.body.collaborators : [];
+
+	const newlyAddedCollaborators = _.differenceWith(newCollaborators, oldCollaborators, _.isEqual);
+	const removedCollaborators = _.differenceWith(oldCollaborators, newCollaborators, _.isEqual);
+
+	const collaboratorPromises = [];
+
+	newlyAddedCollaborators.forEach((collaborator) => {
+		collaboratorPromises.push(
+			new UserCollectionCollaborator({
+				collectionId: newCollection.get('id'),
+				editorId: collaborator.id
+			}).save(null, {method: 'insert'})
+		);
+	});
+	removedCollaborators.forEach((collaborator) => {
+		collaboratorPromises.push(
+			new UserCollectionCollaborator({})
+				.where('collection_id', newCollection.get('id'))
+				.where('editor_id', collaborator.id)
+				.destroy()
+		);
+	});
```

Comment:
``` This method (along with Collection#detach and Collection#updatePivot) are mixed in to a Collection when returned by a belongsToMany relation. ```

Here I used `hasMany` relation. [UserCollectionModel](https://github.com/bookbrainz/bookbrainz-data-js/blob/fc02949c3426a7eb213ad68df6c544045709fbf9/src/models/userCollection.js#L25)

Also `UserCollection().collaborators.attach()` is giving error : `attach is not a function` because of `hasMany` relation

Reply:
Well, in my experience it should work for both sides of the relationship, but maybe I'm getting it wrong. 
It should be `UserCollection().collaborators().attach()` with parenthesis after collaborators; Wanna try it? It'll make your job a bit easier.

Reply:
Oh wait! oops. I was trying it wrong it think. I was doing `UserCollection.collaborators.....`

Reply:
Umm no `attach` works only with `belongsToMany` relation. :(